vantage6-server/vantage6/server/resource/study.py

```diff
@@ -0,0 +1,879 @@
+        if errors:
+            return {
+                "msg": "Request body is incorrect",
+                "errors": errors,
+            }, HTTPStatus.BAD_REQUEST
+
+        name = data["name"]
+        if db.Study.exists("name", name):
+            return {
+                "msg": f"Study with name '{name}' already exists!"
+            }, HTTPStatus.BAD_REQUEST
+
+        if not self.r.can_for_col(P.CREATE, data["collaboration_id"]):
+            return {
+                "msg": "You lack the permission to do that!"
+            }, HTTPStatus.UNAUTHORIZED
+
+        organizations_in_collab = self._check_membership_collaboration(
+            data["organization_ids"], data["collaboration_id"]
+        )
+        if not organizations_in_collab:
+            return {
+                "msg": "Not all organizations are part of the collaboration!"
+            }, HTTPStatus.BAD_REQUEST
+
+        study = db.Study(
+            name=name,
+            collaboration_id=data["collaboration_id"],
+            organizations=[
+                db.Organization.get(org_id)
+                for org_id in data["organization_ids"]
+                if db.Organization.get(org_id)
+            ],
+        )
+
+        study.save()
+        return study_with_orgs_schema.dump(study), HTTPStatus.CREATED
+
+
+class Study(StudyBase):
+    """Resource for /api/study/<int:id>."""
+
+    @only_for(("user", "node", "container"))
+    def get(self, id):
+        """Get study
+        ---
+        description: >-
+          Returns the study with the specified id.\n
+
+          ### Permission Table\n
+          |Rulename|Scope|Operation|Assigned to Node|Assigned to Container|
+          Description|\n
+          | -- | -- | -- | -- | -- | -- |\n
+          |Study|Global|View|❌|❌|All studies|\n
+          |Study|Collaboration|View|❌|❌|Studies which are part of collaborations
+          that your organization participates in|\n
+          |Study|Organization|View|✅|✅|Studies in which
+          your organization participates |\n\n
+
+          Accessible to users.
+
+        parameters:
+          - in: path
+            name: id
+            schema:
+              type: integer
+              minimum: 1
+            description: Study id
+            required: true
+          - in: query
+            name: include
+            schema:
+              type: array
+              items:
+                type: string
+            description: Include 'organizations' to include the organizations
+              within the study.
+
+        responses:
+          200:
+            description: Ok
+          404:
+            description: Study with specified id is not found
+          401:
+            description: Unauthorized
+
+        security:
+          - bearerAuth: []
+
+        tags: ["Collaboration"]
+        """
+        study = db.Study.get(id)
+
+        # check that study exists, unlikely to happen without ID
+        if not study:
```

Comment:
Unlikely but not impossible? I think there might be servers that do not use the study concept at all? I think best to check first if an `id` has been given and then if there is no study. 

```suggestion
        if id and not study:
```

In case en empty list is returned (when no `id` has been specified) seems valid to me. Now the users will get an error that the `id=None` does not exist.

Reply:
This was an old, now irrelevant comment that arised because I copied the endpoints from `/collaboration` and then modified them to become the study endpoints. I now removed the 'unlikely to happen without ID' from both. An id is always specified in GET requests (otherwise it doesn't find its way to this endpoint).

It is indeed possible at present that ID is not a valid positive integer - we should do input schema validation on GET and DELETE requests as well (it's currently only for PATCH and POST). There is already an issue for this somewhere.